core/main/src/Utils/ColorUtils.ts

```diff
@@ -479,16 +479,33 @@ export class ColorUtils {
         };
     }
 
-    static replaceColorSvg(image: IImage, color: IHsl, opacity: number): string {
-        if (!image.svgData) {
-            return "";
+    static replaceColorSvg(imageShape: IImage, color: IHsl, opacity: number): string {
+        if (!imageShape) {
+            throw new Error("Error tsParticles - No image provided.");
+        }
+
+        const { source, svgData } = imageShape;
+        if (!source || !svgData) {
+            throw new Error("Error tsParticles - No image.src");
+        }
+
+        const imageType = source.substr(source.length - 3);
+        if (imageType !== "svg") {
+            throw new Error("Error tsParticles - replaceColor property provided to non-.svg image.src");
         }
 
         /* set color to svg element */
-        const svgXml = image.svgData;
-        const rgbHex = /#([0-9A-F]{3,6})/gi;
+        if (svgData.includes("fill")) {
+            const currentColor = /(#(?:[0-9a-f]{2}){2,4}|(#[0-9a-f]{3})|(rgb|hsl)a?\((-?\d+%?[,\s]+){2,3}\s*[\d.]+%?\)|currentcolor)/i;
+            return svgData.replace(currentColor, () => ColorUtils.getStyleFromHsl(color, opacity));
+        }
+
+        const coloredSvgData = `${svgData.substring(0, svgData.indexOf(">"))} fill="${ColorUtils.getStyleFromHsl(
```

Comment:
This is not clear to me too. Can you give more details? A sample is enough I think.

Reply:
If there is no fill on the SVG, it will split the string at the the end of the first tag <svg> and give it a fill property containing the color to be replaced.